isis/src/qisis/objs/JigsawSetupDialog/JigsawSetupDialog.ui

```diff
@@ -330,69 +330,98 @@ Must be positive.</string>
                   <bool>false</bool>
                  </property>
                  <layout class="QGridLayout" name="gridLayout_14">
-                  <item row="7" column="0" alignment="Qt::AlignRight">
-                   <widget class="QLabel" name="maximumLikelihoodModel3Label">
+                  <item row="6" column="1" colspan="2">
+                   <widget class="QComboBox" name="maximumLikelihoodModel1ComboBox">
+                    <property name="sizePolicy">
+                     <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+                      <horstretch>0</horstretch>
+                      <verstretch>0</verstretch>
+                     </sizepolicy>
+                    </property>
+                    <item>
+                     <property name="text">
+                      <string>NONE</string>
```

Comment:
The elements of the *maximumLikelihoodModel1-3ComboBox's* can cause a PROGRAMMER ERROR to occur, depending on what the user selects.

If you go into the JigsawSetupDialog, select the "HUBER MODIFIED" from the 1st model, then hit OK. The following error occurs:

```
**PROGRAMMER ERROR** Unknown maximum likelihood model name HUBER MODIFIED.
```

This is related to the *MaxiumLikelihoodWFunctions::stringToModel(QString)* method that is called at *JigsawSetupDialog::bundleSettings()* (lines ~600 to ~615). The "HUBER MODIFIED" value does not have a valid conversion defined in *MaxiumLikelihoodWFunctions::stringToModel(QString)*.

You could fix this by changing "HUBER MODIFIED" to "HUBER_MODIFIED".

Another way you could fix it is to programmatically set the elements of the maximumLikelihoodModel comboboxes in the JigsawSetupDialog constructor. If you set the elements there, you can use enum defined in the MaximumLikelihoodWFunctions class to determine the string. That way, the values in these comboboxes are decoupled from JigsawSetupDialog.

You could do this in the constructor with something like:

```
JigsawSetupDialog::JigsawSetupDialog() {
...

// outlier rejection
...
QStringList model1;
model1.append("NONE");
model1.append(MaximumLikelihoodWFunctions::modelToString(Model::Huber);
model1.append(MaximumLikelihoodWFunctions::modelToString(Model::HuberModified);

QStringList model2(model1);
model1.append(MaximumLikelihoodWFunctions::modelToString(Model::Welsch);
model1.append(MaximumLikelihoodWFunctions::modelToString(Model::Chen);

QStringList &model3 = model2;

m_ui->maximumLikelihoodModel1ComboBox.insertItems(0, model1);
m_ui->maximumLikelihoodModel2ComboBox.insertItems(1, model2);
m_ui->maximumLikelihoodModel2ComboBox.insertItems(2, model3);